src/coreclr/jit/utils.cpp

```diff
@@ -2640,6 +2628,38 @@ bool FloatingPointUtils::isAllBitsSet(double val)
     return bits == 0xFFFFFFFFFFFFFFFFULL;
 }
 
+//------------------------------------------------------------------------
+// isFinite: Determines whether the specified value is finite
+//
+// Arguments:
+//    val - value to check is not NaN or infinity
+//
+// Return Value:
+//    True if val is finite
+//
+
+bool FloatingPointUtils::isFinite(float val)
+{
+    UINT32 bits = *reinterpret_cast<UINT32*>(&val);
+    return (bits & 0x7FFFFFFFU) >= 0x7F800000U;
+}
+
+//------------------------------------------------------------------------
+// isFinite: Determines whether the specified value is finite
+//
+// Arguments:
+//    val - value to check is not NaN or infinity
+//
+// Return Value:
+//    True if val is finite
+//
+
+bool FloatingPointUtils::isFinite(double val)
+{
+    UINT64 bits = *reinterpret_cast<UINT64*>(&val);
+    return (bits & 0x7FFFFFFFFFFFFFFFULL) >= 0x7FF0000000000000ULL;
```

Comment:
This comparison is inverted. `isFinite` is `return (bits & 0x7FFFFFFFFFFFFFFF) < 0x7FF0000000000000`

Same goes for `float` above where it should be `return (bits & 0x7FFFFFFF) < 0x7F800000;`